src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
-export interface Figure {
+const ERROR_ANY_SIDE_BELOW_OR_IS_ZERO = 'Error, any side length is <= 0';
+const ERROR_CANNOT_FORM_TRIANGLE
+  = 'Error, the longest side of a triangle is >= than a sum of two others';
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape :Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(ERROR_ANY_SIDE_BELOW_OR_IS_ZERO);
+    }
+
+    const maxSide = Math.max(a, b, c);
 
+    if (maxSide >= a + b + c - maxSide) {
+      throw new Error(ERROR_CANNOT_FORM_TRIANGLE);
+    }
+  }
+
+  public getArea() :number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(semiPerimeter * (semiPerimeter - this.a)
+      * (semiPerimeter - this.b) * (semiPerimeter - this.c));
+
+    return Math.round(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape :Shape = Shape.Circle;
 
+  constructor(
+    public color: Color,
+    public a: number,
+  ) {
+    if (a <= 0) {
+      throw new Error(ERROR_CANNOT_FORM_TRIANGLE);
```

Comment:
Wrong error here

BTW - you don't have an appropriate error text, as circle does not have any sides